move Functionalize dispatch key closer to backends

bdhirsh opened the pull request:
Need this to get functionalize to work with backends (LTC/XLA). Now that we can kill the `DECOMPOSE_FUNCTIONAL` code in functorch (see https://github.com/pytorch/functorch/pull/814), this should be ok to land once that PR merges.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #75527
* #76358
* #77714
* #77713
* __->__ #77132
* #77716

bdhirsh committed:
move Functionalize dispatch key closer to backends

[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"




[ghstack-poisoned]

zou3519 reviewed (approved):
lgtm, but please check that functorch functionalize tests pass after this change

bdhirsh committed:
Update on "move Functionalize dispatch key closer to backends"

Need this to get functionalize to work with backends (LTC/XLA). Now that we can kill the `DECOMPOSE_FUNCTIONAL` code in functorch (see https://github.com/pytorch/functorch/pull/814), this should be ok to land once that PR merges.




[ghstack-poisoned]
